### PR TITLE
Added code for chathooks

### PR DESCRIPTION
**Pull request by emma-miler**
This may not seem like much to a passing observer, but this commit took me 30 hours of blood, sweat, tears, IDA debugging, server crashes, and insanity.
Should be paired with a commit in Mods

### CODE REVIEW

**Comment by abarichello** [comment]
#69 nice

**Comment by emma-miler** [comment]
nice

**Comment by x3Karma** [comment]
nice

File: NorthstarDedicatedTest/serverauthentication.cpp [hunk 1]
```diff
@@ -516,7 +518,32 @@ bool ProcessConnectionlessPacketHook(void* a1, netpacket_t* packet)
...
+SQRESULT setMessage(void* sqvm)
+{
+	currentMessage = ServerSq_getstring(sqvm, 1);
+	currentPlayerId = ServerSq_getinteger(sqvm, 2);
+	currentChannelId = ServerSq_getinteger(sqvm, 3);
+	shouldBlock = ServerSq_getbool(sqvm, 4);
+	return SQRESULT_NOTNULL;
+}
+
+void CServerGameDLL__OnRecievedSayTextMessageHook(void* self, unsigned int senderClientIndex, const char* message, int channelId)
```
Comment (abarichello) [inline NorthstarDedicatedTest/serverauthentication.cpp:546]: Can you fix all occurrences of this _received_ typo please?

File: NorthstarDedicatedTest/serverauthentication.cpp [hunk 2]
```diff
@@ -532,10 +559,26 @@ void CServerGameDLL__OnRecievedSayTextMessageHook(void* self, unsigned int sende
 
 	g_ServerAuthenticationManager->m_additionalPlayerData[sender].sayTextLimitCount++;
 
-	// todo: could censor messages here if we have a banned word list, we do not currently have one of these
-	// could possibly make this call a script codecallback, or smth
+	bool shouldDoChathooks = strstr(GetCommandLineA(), "-enableChatHooks");
```
Comment (abarichello) [inline NorthstarDedicatedTest/serverauthentication.cpp:562]: cmdline arguments are lowercase, i think `-enablechathooks` or something like that makes more sense

File: NorthstarDedicatedTest/serverauthentication.cpp (hunk 1 above)
Comment (Coopyy) [reply NorthstarDedicatedTest/serverauthentication.cpp:546]: i before e except after c smh

**Pull request merged**